lib/programmemory.h

```diff
@@ -96,12 +96,21 @@ struct ExprIdToken {
     };
 };
 
-struct ProgramMemory {
+struct CPPCHECKLIB ProgramMemory {
     using Map = std::unordered_map<ExprIdToken, ValueFlow::Value, ExprIdToken::Hash>;
 
-    ProgramMemory() = default;
+    ProgramMemory() : mValues(new Map()), mCopyValues(false) {}
 
-    explicit ProgramMemory(Map values) : mValues(std::move(values)) {}
+    explicit ProgramMemory(std::shared_ptr<Map> values) : mValues(std::move(values)), mCopyValues(true) {}
+
+    ProgramMemory(const ProgramMemory &other) : mValues(other.mValues), mCopyValues(true) {}
+
+    ProgramMemory &operator=(const ProgramMemory &other)
+    {
+        mValues = other.mValues;
+        mCopyValues = true;
+        return *this;
+    }
```

Comment:
Just use the default-generated copy constructor and assignment.

Reply:
Without the need for `mCopyValues` I can do taht now.

It would have also required the move constructor/assignment.

Reply:
Done.